src/server/test/web/csvPipelineTest.js

```diff
@@ -518,23 +557,40 @@ for (let fileKey in testCases) {
 			let expectedFile = `${fileKey}Expected.csv`;
 			let expectedPath = `${__dirname}/csvPipeline/${expectedFile}`;
 			let expectedBuffer = fs.readFileSync(expectedPath);
+			const conn = testDB.getConnection();
 			for (let index = 0; index < numUploads; index++) {
 				// It would be nice to put a mocha.describe inside the loop to tell the upload starting
 				// but that breaks the tests.
 				// Each set of uploads must be in one mocha because the DB is reset with each test.
+				if (index == 0 && testCases[fileKey].hasOwnProperty('createMeter') && testCases[fileKey]['createMeter']) {
+					// The key to create meter is present and true so create the meter if first request.
+					const meter = new Meter(
+						undefined, // id
+						fileKey, // name
+						undefined, // url
+						false, // enabled
+						false, // displayable
+						Meter.type.OTHER, // type
+						null, // timezone
+					)
+					meter.insert(conn);
+				}
 				let inputFile = testCases[fileKey]['fileName'][index];
 				let inputPath = `${__dirname}/csvPipeline/${inputFile}`;
 				let inputBuffer = fs.readFileSync(inputPath);
 				let evalString = `${testCases[fileKey]['chaiRequest'][index]}.attach('csvfile', inputBuffer, '${inputPath}')`;
 				// TODO It would be nice if this was not an eval. Tried a function with closure but could not get it to work as did not find chai.
 				const res = await eval(evalString);
 				expect(res).to.have.status(testCases[fileKey]['responseCode'][index]);
-				expect(res).to.be.html;
 				// OED returns a string with messages that we check it is what was expected.
-				expect(res.text).to.equal(testCases[fileKey]['responseString'][index]);
+				try {
```

Comment:
A few of thoughts on this:
1. The check for html was removed but a json check did not replace it.
2. The fact that the test needed to be modified is an indication that you changed the expected response. These responses have been published and sites have a reason to expect they will not change unless absolutely needed. They were set up as html for a reason and now this one is not html. I'm hoping the old behavior can easily be put back. Maybe you can not put a success code in the response as JSON and look at the response code. It looks as if that changes then the added item to via the JSON string is not needed.
3. Usually you have no need for a try/catch like this in a test. I think something unusual/undesirable is happening here.
I have not spent a lot of time on this so maybe I've missed something. We might need to discuss.